[issue] #74 Dark Mode

aashdeepbrar opened the pull request.

lvlingxiao1 committed:
add dark mode switch

aashdeepbrar committed:
Fixing up dark mode

aashdeepbrar committed:
Fixing slide overlap with buttons

christinetg committed:
more dark mode adjustments

christinetg committed:
Merge branch 'dark-mode' of https://github.com/lvlingxiao1/slidechat into dark-mode

christinetg committed:
text field background

christinetg committed:
removed text bg color variable

aashdeepbrar committed:
Fixing syntax

aashdeepbrar committed:
Merge branch 'dark-mode' of https://github.com/lvlingxiao1/slidechat into dark-mode

aashdeepbrar committed:
Adding more dark mode stuff

aashdeepbrar committed:
Removing unused variables

lvlingxiao1 commented:
It looks good!

A few things we can improve:
It seems the colour of the Material UI switch is affected when it's not turned on,
![image](https://user-images.githubusercontent.com/43081785/108144881-842fd800-7098-11eb-8447-6cf30836ba17.png)
Let's still make it a white thumb and gray track as before, I think that looks better.
Also, the thumb is a little hard to see in dark mode. Here is what Material UI switch looks like
![image](https://user-images.githubusercontent.com/43081785/108145452-74fd5a00-7099-11eb-9fd2-3ca97a12b015.png)

The background in dark mode is (almost) pitch black right now. It might make the contrast too big. Try #2b2b2b for primary and #252525 for background and see if you like it.

Disabled buttons might look too similar to active buttons
![image](https://user-images.githubusercontent.com/43081785/108145677-d45b6a00-7099-11eb-86cf-8f9474625c2c.png)

aashdeepbrar committed:
Addressing CR comment

christinetg committed:
toggle switch colours

christinetg committed:
switch colours

lvlingxiao1 commented:
I still feel maybe a slightly bright background could be useful
buttons with colour "primary" is a little not so stand out from the background (e.g. "start a new discussion", "send"). When hovering over, they don't give enough feedback either (in light mode there is a drop shadow), brighten the background might the shadow more visible (or we can make some other mechanisms... like brighten the button when hovered)
background with #333333
![image](https://user-images.githubusercontent.com/43081785/108251108-06fe7480-7125-11eb-9f3a-d20d7e9eda17.png)

christinetg committed:
light dark mode

christinetg committed:
button colour

lvlingxiao1 committed:
reorganize variables
change primary buttons to brighter colour
fix input border colour

lvlingxiao1 committed:
Add styles for links
Fix page number input box style in dark mode
Change "x seconds ago" to "Just now" to above issue of negative seconds ago caused by slightly out of sync clocks

lvlingxiao1 committed:
slightly brighten the background
